packages/lesswrong/components/events/modules/VirtualProgramCard.tsx

```diff
@@ -0,0 +1,199 @@
+    boxShadow: "0 1px 3px rgba(0, 0, 0, 0.1)",
+    [theme.breakpoints.down('xs')]: {
+      maxWidth: '100vw'
+    }
+  },
+  introVPCard: {
+    background: "linear-gradient(rgba(0, 87, 102, 0.7), rgba(0, 87, 102, 0.7)), url('https://res.cloudinary.com/cea/image/upload/w_374,h_373,c_fill,q_auto,f_auto/Event/pz3xmsm63xl8thlyt2up.jpg')",
+    padding: '50px 24px',
+    '& .VirtualProgramCard-eventCardDescription': {
+      opacity: 1,
+      lineHeight: '1.8em',
+      marginTop: 30
+    },
+    '& .VirtualProgramCard-eventCardDeadline': {
+      marginTop: 30
+    }
+  },
+  cardLink: {
+    '&:hover': {
+      opacity: '0.9'
+    },
+    '&:hover .VirtualProgramCard-eventCardDeadline': {
+      borderBottom: '2px solid white'
+    }
+  },
+  cardSection: {
+    display: 'flex',
+    width: 373,
+    height: 243,
+    padding: 20,
+    overflow: 'hidden'
+  },
+  inDepthSection: {
+    background: "linear-gradient(rgba(0, 87, 102, 0.7), rgba(0, 87, 102, 0.7)), url('https://res.cloudinary.com/cea/image/upload/w_374,h_243,c_fill,q_auto,f_auto/Event/f2cbeqvjyhyl6rhhzdsu.jpg')",
+    clipPath: 'polygon(0 0, 100% 0, 100% 54%, 0 100%)'
+  },
+  precipiceSection: {
+    background: "linear-gradient(rgb(168, 114, 51, 0.5), rgb(168, 114, 51, 0.5)), url('https://res.cloudinary.com/cea/image/upload/w_374,h_243,c_fill,q_auto,f_auto/Event/xfhrtorwdxxmplaofqa8.jpg')",
+    clipPath: 'polygon(0 46%, 100% 0, 100% 100%, 0 100%)',
+    position: 'absolute',
+    bottom: 0,
+    alignItems: 'flex-end',
+    textAlign: 'right'
+  },
+  eventCardTime: {
+    ...theme.typography.commentStyle,
+    fontSize: 14,
+    color: 'white'
+  },
+  eventCardTitle: {
+    ...theme.typography.headline,
+    color: 'white',
+    fontSize: 22,
+    marginTop: 8,
+    marginBottom: 0
+  },
+  eventCardLocation: {
+    ...theme.typography.commentStyle,
+    opacity: '0.7',
+    color: 'white',
+    fontSize: 14,
+    marginTop: 8,
+  },
+  eventCardDescription: {
+    ...theme.typography.commentStyle,
+    opacity: '0.7',
+    color: 'white',
+    fontSize: 14,
+    lineHeight: '1.5em',
+    marginTop: 10,
+  },
+  eventCardDeadline: {
+    ...theme.typography.commentStyle,
+    display: 'inline-block',
+    fontWeight: 'bold',
+    color: 'white',
+    fontSize: 16,
+    paddingBottom: 5,
+    marginTop: 10,
+    borderBottom: '2px solid transparent'
+  },
+}))
+
+
+const VirtualProgramCard = ({program, classes}: {
+  program: string,
+  classes: ClassesType,
+}) => {
+  const { captureEvent } = useTracking()
+  
+  // find the next deadline for applying to the Intro VP, which is the last Sunday of every month
+  let sunday = moment().day(0)
+  _.range(5).forEach(() => {
+    const nextSunday = moment(sunday).add(1, 'week')
+    // needs to be in the future
+    if (sunday.isBefore(moment(), 'day')) {
+      sunday = nextSunday
+    }
+    // needs to be the last Sunday of the month
+    if (nextSunday.month() === sunday.month()) {
+      sunday = nextSunday
+    }
+  })
```

Comment:
Two cents: repeatedly mutating `sunday` is confusing to me. I *think* the following is equivalent and more intuitive to my brain
```
_.range(5).map(x => moment(sunday).add(x, 'week'))
.filter(nextSunday => nextSunday.month() === sunday.month())
.last()

Reply:
I don't think that quite works, because it doesn't handle moving into the next month. Ex. if today is Monday Feb 28, then `sunday` would be Feb 27, and I think `last()` would return Feb 27 because all the other Sundays are in future months.

I could separately check if `moment(sunday).add(1, 'week')` is in the same month or not - do you think this is clearer?
```
const month = moment(sunday).add(1, 'week').month() === sunday.month() ? sunday.month() : (sunday.month() + 1)%12
const deadline = _.range(5).map(x => moment(sunday).add(x, 'week'))
    .filter(nextSunday => nextSunday.month() === month)
    .last()
```

I think I find this slightly more confusing than my code, because of the logic duplicated in `month` and `deadline`.

Maybe the reason my code is confusing is because it's meant to be a `while` loop instead of a `forEach`, but I went with the latter to avoid possibly causing an infinite loop by accident. It's trying to move to the next Sunday "while this Sunday is either in the past or is not the last Sunday of the month". Let me know if you think I should just use a `while` loop, if that would be clearer.

Reply:
I made my contribution as well. I found it confusing at first read, I think for the reason Sarah said. I often find fancy functional tricks harder to quickly read, so I vote for my way, but ¯\\\_(ツ)_/¯

Reply:
Ah okay, I do actually find that clearer because to my mind you are actually trying to do two separate things (determine the month of the next Sunday and then determine the last Sunday of that month), so putting them on two separate lines makes sense.

If you decide to go the `while` route, I would suggest using `reduce` instead. This avoids side effects and, to your point, is also guaranteed to terminate. (Although I see that JP suggested going in an entirely different direction, so maybe this is just too much of my personal preference.)

Reply:
Ok I'll go with JP's version, I can see why it's clearest. But I like `forEach` when there is no need to end early, so get used to seeing more of those. (I also like `reduce` but it doesn't seem like a good fit for this case.)